### ISSUE TITLE
[test] add a `IPAdapterTexterMixin`

### ISSUE THREAD

**Issue by yiyixuxu** [huggingface/diffusers#6829]
I think we should add a `IPAdapterTexterMIxin` and run fast tests on every pipeline that supports IP-Adapter, so that we will be able to catch breaking changes like this https://github.com/huggingface/diffusers/pull/6828

some thoughts:
* we can add a `IPAdapterTexterMIxin` class to `test_pipelines_common.py` https://github.com/huggingface/diffusers/blob/9cc59ba0891c3ccbebfcbb6784f4dde2d5bd8556/tests/pipelines/test_pipelines_common.py#L62C7-L62C32
* we can then add `IPAdapterTexterMIxin` as a base class to the fast tests of all the pipelines support ip-adapter, for example,add here for `StableDiffusionXLPipelineFastTests` (https://github.com/huggingface/diffusers/blob/9cc59ba0891c3ccbebfcbb6784f4dde2d5bd8556/tests/pipelines/stable_diffusion_xl/test_stable_diffusion_xl.py#L59)
* we can either make a dummy repo for ip-adapter ( i.e.`hf-internal-testing/tiny_random_ip_adapter`) or use this method https://github.com/huggingface/diffusers/blob/9cc59ba0891c3ccbebfcbb6784f4dde2d5bd8556/tests/models/unets/test_models_unet_2d_condition.py#L59C34-L59C39 to create `dummy_ip_adapter_state_dict` based on `self.get_dummy_components()['unet']`
*  add a simple test: 
   * the output of the pipeline without ip-adapter should be the same as pipeline with ip-adapter but with ip_adapter_scale = 0
   * the output of pipeline without ip-adapter should NOT be same as pipeline with ip-adapter with ip_adapter_scale = 1
* the tests should only run if the pipeline's  `__call__` method has the `ip_adapter_image` argument

cc @DN6 @sayakpaul let me know what you guys think!

**Comment by sayakpaul** [comment]
100 percent agree.

**Comment by DN6** [comment]
Sounds good to me 👍🏽

**Comment by UmerHA** [comment]
Hey @yiyixuxu! I'd give this a go, if okay for you!

**Comment by yiyixuxu** [comment]
@UmerHA 
we already have a PR open and very close to merge for this now:) https://github.com/huggingface/diffusers/pull/6862

checkout other issues here to see if any of them interest you:)
https://github.com/huggingface/diffusers/issues?q=is%3Aopen+is%3Aissue+label%3Acontributions-welcome
